spec/models/results/web_response_parser_spec.rb

```diff
@@ -7,237 +7,527 @@
-        "1" => {
-          id: "",
-          type: "AnswerGroupSet",
-          questioning_id: form.c[1].id,
-          relevant: "true",
-          children: {
-            "0" => web_answer_group_hash(form.c[1].id, instance_one_answers),
-            "1" => web_answer_group_hash(form.c[1].id, instance_two_answers)
+      context "new value" do
+        let(:new_answers) do
+          {
+            "0" => web_answer_hash(form.c[0].id, value: "A", id: response.root_node.c[0].id),
+            "1" => web_answer_hash(form.c[1].id, value: "B", id: response.root_node.c[1].id),
+            "2" => web_answer_hash(form.c[2].id, value: "Z", id: response.root_node.c[2].id)
           }
-        }
-      }
-    end
-    let(:instance_one_answers) do
-      {
-        "0" => web_answer_hash(form.c[1].c[0].id, value: "B"),
-        "1" => web_answer_hash(form.c[1].c[1].id, value: "C")
-      }
+        end
+
+        it "updates value appropriately" do
+          expect_root(tree, form)
+          expect_children(tree, %w[Answer Answer Answer], form.sorted_children.map(&:id), %w[A B Z])
+        end
+      end
+
+      context "destroy flag set to true" do
+        let(:new_answers) do
+          {
+            "0" => web_answer_hash(form.c[0].id, value: "A", id: response.root_node.c[0].id),
+            "1" => web_answer_hash(
+              form.c[1].id,
+              value: "B",
+              id: response.root_node.c[1].id,
+              _destroy: "true"
+            ),
+            "2" => web_answer_hash(form.c[2].id, value: "C", id: response.root_node.c[2].id)
+          }
+        end
+
+        it "updates destroy appropriately in in-memory tree" do
+          expect_root(tree, form)
+          expect_children(tree, %w[Answer Answer Answer], form.sorted_children.map(&:id), %w[A B C])
+          expect(tree.c[1]._destroy).to be true
+        end
+      end
+
+      context "relevant set to false" do
+        let(:new_answers) do
+          {
+            "0" => web_answer_hash(form.c[0].id, value: "A", id: response.root_node.c[0].id),
+            "1" => web_answer_hash(
+              form.c[1].id,
+              value: "B",
+              id: response.root_node.c[1].id,
+              relevant: "false"
+            ),
+            "2" => web_answer_hash(form.c[2].id, value: "C", id: response.root_node.c[2].id)
+          }
+        end
+
+        it "updates relevant appropriately in in-memory tree" do
+          expect_root(tree, form)
+          expect_children(tree, %w[Answer Answer Answer], form.sorted_children.map(&:id), %w[A B C])
+          puts "node object id in spec #{tree.children[1]}"
+          expect(tree.children[1].relevant).to eq false
+        end
+      end
     end
-    let(:instance_two_answers) do
-      {
-        "0" => web_answer_hash(form.c[1].c[0].id, value: "D"),
-        "1" => web_answer_hash(form.c[1].c[1].id, value: "E")
-      }
+
+    context "repeat groups" do
```

Comment:
Do we still need this one if we are covering the nested repeat case? This file is huge and (unavoidably) kind of hard to parse so if we can shrink it a bit that might be nice.

Reply:
I think it'll be nice to have separate specs if there's an issue with nested in particular. I agree it's an unfortunately large spec.

Reply:
For parsing's sake, we could move the update specs to another file. That would make it less daunting and easier to parse the lets and context blocs.